granulate_utils/metrics/yarn/utils.py

```diff
@@ -158,8 +159,14 @@ def detect_yarn_config(*, logger: Union[logging.Logger, logging.LoggerAdapter])
     """
     if yarn_home_dir := find_yarn_home_dir(logger=logger):
         logger.debug(f"found YARN home dir: {yarn_home_dir}")
-        yarn_site_xml_file = Path(yarn_home_dir).joinpath(RELATIVE_YARN_SITE_XML_PATH)
-        return read_config_file(yarn_site_xml_file, logger=logger)
+        try:
+            yarn_site_xml_file = glob.glob(f"{yarn_home_dir}/**/yarn-site.xml", recursive=True)[0]
```

Comment:
Log a warning if more than one is found, this is not expected and may cause errors.